pkg/ruler/ruler.go

```diff
@@ -74,14 +74,21 @@ type Config struct {
 	StoreConfig RuleStoreConfig `yaml:"storage"`
 	// Path to store rule files for prom manager.
 	RulePath string `yaml:"rule_path"`
-
+	// Sets the Authorization header on every request to the Alertmanager with the configured username.
```

Comment:
Sadly, I had to introduce more flags given we no longer can assume a scheme, path, username and password based on URL like we did when it was just one. The good news is that this aligns with what the users of Prometheus would expect as this what the Alertmanager configuration supports.